### PR TITLE
Fixed files so that snake_case variables are camelCase, going along with the ESLint guidelines.

### PR DESCRIPTION
**Pull request by ultra-programmer**
In accordance with ESLint, the files in musicbrainz-server have been edited to have their variables follow camelCase .

Command used: `./script/check_eslint_rule 'camelcase: [warn, {properties: never, allow: [ "^(N_)?l[np]?(_[a-z_]+)?$" ]}]' root/`

### CODE REVIEW

File: root/static/scripts/common/utility/formatDatePeriod.js [hunk 1]
```diff
@@ -12,32 +12,32 @@ import ko from 'knockout';
 import formatDate from './formatDate';
 
 function formatDatePeriod<+T: {...DatePeriodRoleT, ...}>(entity: T) {
-  let {begin_date, end_date, ended} = entity;
+  let {beginDate, endDate, ended} = entity;
```
Comment (ABCbum) [inline root/static/scripts/common/utility/formatDatePeriod.js]: And also here the [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) requires you to give precise property name which means `beginDate` in this case is actually `undefined`.

File: root/static/scripts/common/MB/Control/Autocomplete.js [hunk 2]
```diff
@@ -984,9 +984,9 @@ MB.Control.autocomplete_formatters = {
             }
         };
 
-        if (item.related_entities) {
-            entityRenderer(l('Performers'), item.related_entities.performers);
-            entityRenderer(l('Location'), item.related_entities.places);
+        if (item.relatedEntities) {
+            entityRenderer(l('Performers'), item.relatedEntities.performers);
+            entityRenderer(l('Location'), item.relatedEntities.places);
```
Comment (mwiencek) [inline root/static/scripts/common/MB/Control/Autocomplete.js]: This should be reverted, since `related_entities` is the name of the property on the object, which comes from the server.

File: root/static/scripts/common/utility/formatDatePeriod.js (hunk 1 above)
Comment (mwiencek) [reply root/static/scripts/common/utility/formatDatePeriod.js]: Yep, these can't be changed like this either. You'll need to define `beginDate` and `endDate` as aliases in the destructuring assignment here... see the section "Assigning to new variable names" in the article @ABCbum linked. :)

File: root/static/scripts/common/utility/formatDatePeriod.js [hunk 3]
```diff
@@ -12,32 +12,32 @@ import ko from 'knockout';
...
   }
 
-  if (begin_date === end_date) {
-    return begin_date;
+  if (beginDate === endDate) {
+    return beginDate;
   }
 
-  if (begin_date && end_date) {
-    return texp.l('{begin_date} \u2013 {end_date}', {begin_date, end_date});
```
Comment (mwiencek) [inline root/static/scripts/common/utility/formatDatePeriod.js]: Revert the change to the translated string here (we use snake_case for those, and changing it means it'd have to be re-translated by our users). The second argument (the string substitutions) should look like `{begin_date: beginDate, end_date: endDate}`.

File: root/static/scripts/common/utility/formatDatePeriod.js [hunk 4]
```diff
@@ -12,32 +12,32 @@ import ko from 'knockout';
...
+  if (beginDate && endDate) {
+    return texp.l('{beginDate} \u2013 {endDate}', {beginDate, endDate});
```
Comment (mwiencek) [inline root/static/scripts/common/utility/formatDatePeriod.js]: See previous comment

File: root/static/scripts/common/utility/formatDatePeriod.js [hunk 5]
```diff
@@ -12,32 +12,32 @@ import ko from 'knockout';
...
   }
 
-  if (!begin_date) {
-    return texp.l('\u2013 {end_date}', {end_date});
+  if (!beginDate) {
+    return texp.l('\u2013 {endDate}', {endDate});
```
Comment (mwiencek) [inline root/static/scripts/common/utility/formatDatePeriod.js]: See previous comment

File: root/static/scripts/common/utility/formatDatePeriod.js [hunk 6]
```diff
@@ -12,32 +12,32 @@ import ko from 'knockout';
...
   }
 
-  if (!end_date) {
+  if (!endDate) {
     return ended
-      ? texp.l('{begin_date} \u2013 ????', {begin_date})
-      : texp.l('{begin_date} \u2013', {begin_date});
+      ? texp.l('{beginDate} \u2013 ????', {beginDate})
+      : texp.l('{beginDate} \u2013', {beginDate});
```
Comment (mwiencek) [inline root/static/scripts/common/utility/formatDatePeriod.js]: See previous comment

File: root/static/scripts/edit/MB/CoverArt.js [hunk 7]
```diff
@@ -204,12 +204,12 @@ MB.CoverArt.file_data_uri = function (file) {
...
 };
 
-MB.CoverArt.sign_upload = function (file, gid, mime_type) {
+MB.CoverArt.sign_upload = function (file, gid, mimeType) {
     var deferred = $.Deferred();
 
     var postfields = $.ajax({
         url: "/ws/js/cover-art-upload/" + gid,
-        data: { mime_type: mime_type },
+        data: { mimeType: mimeType },
```
Comment (mwiencek) [inline root/static/scripts/edit/MB/CoverArt.js]: The data being sent to the server can't be changed here, 'cause the server won't understand it :)
```suggestion
        data: { mime_type: mimeType },
```

File: root/static/scripts/edit/MB/CoverArt.js [hunk 8]
```diff
@@ -349,8 +349,8 @@ MB.CoverArt.FileUpload = function (file) {
 
 
     MB.CoverArt.file_data_uri(file)
-        .done(function (data_uri) {
-            self.data_uri_data(data_uri);
+        .done(function (dataUri) {
+            self.dataUri_data(dataUri);
```
[the remark this answers is not in the document]
Comment (mwiencek) [reply root/static/scripts/edit/MB/CoverArt.js]: `dataUriData` would make more sense but yeah, the declaration needs fixing

File: root/static/scripts/edit/MB/edit.js [hunk 9]
```diff
@@ -9,7 +9,7 @@
 import ko from 'knockout';
 import _ from 'lodash';
 
-import {hex_sha1} from '../../../lib/sha1/sha1';
+import {hexSha1} from '../../../lib/sha1/sha1';
```
Comment (mwiencek) [inline root/static/scripts/edit/MB/edit.js]: You can't import using a different name quite like that, since it's still `hex_sha1` in the file, but you can specify an alias:
```suggestion
import {hex_sha1 as hexSha1} from '../../../lib/sha1/sha1';
```

File: root/static/scripts/release-editor/utils.js [hunk 10]
```diff
@@ -14,7 +14,7 @@
 import ko from 'knockout';
 import _ from 'lodash';
 
-import {rstr_sha1} from '../../lib/sha1/sha1';
+import {rstrSha1} from '../../lib/sha1/sha1';
```
Comment (mwiencek) [inline root/static/scripts/release-editor/utils.js]:
```suggestion
import {rstr_sha1 as rstrSha1} from '../../lib/sha1/sha1';
```

**Comment by ultra-programmer** [comment]
I have added the changes requested by @ABCbum 
I will get to fixing the other issues right away! Apologies for the errors, but thanks so much for the helpful information.

File: root/static/scripts/edit/MB/edit.js (hunk 9 above)
Comment (ultra-programmer) [reply root/static/scripts/edit/MB/edit.js]: Apologies, that was my bad.

**Comment by reosarevok** [comment]
Hi! There's some conflicts there, please fix those and resubmit :)

File: root/static/scripts/guess-case/MB/Control/GuessCase.js [hunk 11]
```diff
@@ -16,7 +16,7 @@ import gc from '../GuessCase/Main';
 import * as modes from '../../modes';
 import MB from '../../../common/MB';
 
-MB.Control.initialize_guess_case = function (type, formPrefix) {
+MB.Control.initializeGuessCase = function (type, formPrefix) {
```
Comment (mwiencek) [inline root/static/scripts/guess-case/MB/Control/GuessCase.js]: There are also some cases in our *.tt files where this needs to be changed now - see `git grep initialize_guess_case` for a complete list.

Comment (ultra-programmer) [reply root/static/scripts/guess-case/MB/Control/GuessCase.js]: I have made the chagnes.

Comment (ultra-programmer) [reply root/static/scripts/guess-case/MB/Control/GuessCase.js]: I have made the changes

File: root/static/scripts/edit/MB/TextList.js [hunk 12]
```diff
@@ -33,17 +33,21 @@ MB.Form.TextList = function (input) {
...
                 .find('div.text-list-row')
                 .last())
             .show()
+<<<<<<< HEAD
+            .find('input.value').attr("name", input + '.' + counter).val(initValue)
+=======
             .find('input.value')
             .attr("name", input + '.' + counter)
             .val(init_value)
+>>>>>>> upstream/master
```
Comment (ABCbum) [inline root/static/scripts/edit/MB/TextList.js]: Hi @ultra-programmer, I think you should resolve the conflict here first.

Comment (ultra-programmer) [reply root/static/scripts/edit/MB/TextList.js]: Oops, well that's embarrassing... Sorry for the error. I will fix it right away!

**Comment by ultra-programmer** [comment]
Hello! My changes seem to not have passed the CircleCI tests. How could I check for and resolve those on my own? Thanks!

**Comment by reosarevok** [comment]
I think that was a temporary issue with the test (the error was about the database not starting). I've restarted the tests to check.

In the meantime, sorry, but I caused another conflict by merging one of my open PRs :/ Can you solve that one too? :)

File: root/static/scripts/common/coverart.js [hunk 13]
```diff
@@ -11,24 +11,23 @@ import $ from 'jquery';
...
-                "title": $e.data("title")
-            })
-            .appendTo(this);
-
+        })
+        .attr({
+            "src": thumbnailUrl,
+            "title": $e.data("title")
+        })
+        .appendTo(this);
```
Comment (mwiencek) [inline root/static/scripts/common/coverart.js]: The rest of the PR looks good, but the indentation here got changed (possibly while resolving conflicts) - can you fix that when you have time? :)
